invenio_openaire/loaders.py

```diff
@@ -277,6 +321,66 @@ def iter_grants(self, as_json=True):
         self._disconnect()
 
 
+class LocalJSONOAIRELoader(BaseOAIRELoader):
+    """Local OpenAIRE dataset loader."""
+
+    def __init__(self, source=None, **kwargs):
+        """Init the loader for local database."""
+        super(LocalJSONOAIRELoader, self).__init__(
+            source,
+            **kwargs)
+
+    def extract_grants(self, source):
+        """Extract grants within a json lines file."""
+        parsed_grants = []
+        with gzip.open(source, 'r') as json_file:
+            json_list = list(json_file)
+
+        for json_grant in json_list:
+            parsed_grants.append(json.loads(json_grant))
+
+        return parsed_grants
```

Comment:
1.  I'd say this might actually be the best place to decide (based on the file extension) if gzip should be used
2. Given that the files can get pretty big, it might be better to have a streaming/yielding approach

```suggestion
        if os.path.splitext(source)[1].lower() == '.gz':  # 1.
            open_func = gzip.open
        else:
            open_func = open
        with open_func(source, 'r') as fp:
            for line in fp:
                yield json_list = list(json.loads(line))  # 2.
```